File: apps/envoygo/lib/eh/eh_review_prefs.dart
import 'package:shared_preferences/shared_preferences.dart';

const _storageKey = 'envoymesh:eh-review-min-files';

/// Minimum changed-file count before auto-opening review (0 = always).
Future<int> getEhReviewMinFiles() async {
  final prefs = await SharedPreferences.getInstance();
  final raw = prefs.getString(_storageKey);
  if (raw == null) return 1;
  final parsed = int.tryParse(raw);
  if (parsed == null || parsed < 0) return 1;
  return parsed;
}

Future<void> setEhReviewMinFiles(int value) async {
  final prefs = await SharedPreferences.getInstance();
  await prefs.setString(_storageKey, '${value < 0 ? 0 : value}');
}

File: apps/social/src/lib/eh-review-prefs.ts
const STORAGE_KEY = "envoymesh:eh-review-min-files";

/** Minimum changed-file count before auto-opening the review panel (0 = always). */
export function getEhReviewMinFiles(): number {
  try {
    const raw = localStorage.getItem(STORAGE_KEY);
    if (raw === null) return 1;
    const parsed = Number(raw);
    if (!Number.isFinite(parsed) || parsed < 0) return 1;
    return Math.floor(parsed);
  } catch {
    return 1;
  }
}

export function setEhReviewMinFiles(value: number): void {
  try {
    localStorage.setItem(STORAGE_KEY, String(Math.max(0, Math.floor(value))));
  } catch {
    // ignore quota / private mode
  }
}
